src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -113,6 +121,20 @@ internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, Ca
             }
         }
 
+        private async Task<IEslintBridgeClient> GetJavaScriptClient()
```

Comment:
I think this method should be wrapped with a lock, to avoid a race condition of multiple file analysis that can happen if the process not started yet or died.
E.g.: open a few files, kill the process, change settings.json --> will spawn multiple analysis that each will attempt to start the server. Perhaps can also happen with first analysis (e.g. open file and immediately save it --> 2 analysis)

Reply:
I don't think we need to worry about the "open then save" case; I think the TextBufferIssueTracker should handle that for us (via the slow-memory-leaking-job-scheduler code).

Changing the settings might cause an issue (when we get round to supporting it).

However, perhaps the `eslintBridgeProcess` should handle it. It's returning a task that could be awaited by multiple callers, so it should protect itself against race conditions so the callers don't have to.